### PR TITLE
chore: run pyupgrade on the repo

### PR DESCRIPTION
**Pull request by cpcloud**
This PR runs pyupgrade on the repo.

### CODE REVIEW

**Comment by jreback** [comment]
wow this is a lot, but needed

File: ibis/backends/base/__init__.py [hunk 1]
```diff
@@ -83,7 +83,7 @@ def current_database(self) -> str | None:
         """
 
     @abc.abstractmethod
-    def list_databases(self, like: str = None) -> List[str]:
+    def list_databases(self, like: str = None) -> list[str]:
```
Comment (jreback) [inline ibis/backends/base/__init__.py:86]: i think this typing sytntax is fine, but requires `from __future__ import annotations` in every module that supports it in order for this to not break 3.7. odd that the CI is passing.

**Review by jreback — CHANGES_REQUESTED** [review]

File: ibis/backends/base/__init__.py (hunk 1 above)
Comment (cpcloud) [reply ibis/backends/base/__init__.py:86]: `from __future__ import annotations` is already in this module. If you expand the file in the GitHub UI you can see it on line 1.

**Comment by cpcloud** [comment]
@jreback `from __future__ import annotations` is already imported where needed (or added by `pyupgrade`). As you note, the CI wouldn't pass if it wasn't.

**Comment by jreback** [comment]
> @jreback `from __future__ import annotations` is already imported where needed (or added by `pyupgrade`). As you note, the CI wouldn't pass if it wasn't.

ahh must have been added previously, ok!

**Comment by jreback** [comment]
not a terrible idea to have pyupgrade as a periodic action, like once a month or something.

**Comment by cpcloud** [comment]
We can run it as a pre commit check as well. Either way regular run is a good idea.

**Pull request merged**